script_based_workflows.Rmd

```diff
@@ -0,0 +1,190 @@
+
+It was acknowledged in the ("What we left out")[https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510#sec027] section that build tools to manage projects when code gets out of date are ideal, but can be substituted with a simple master script called `make.R` that reruns each of the scripts in order.
+
+```{r,eval=FALSE}
+
+# Preprocessing steps
+source("01_data.R")
+source("02_munge.R")
+source("03_histogram.R")
+source("04_regression.R")
+
+# Calls rmarkdown::render on report.Rmd
+source("05_report.R")
+```
+
+The argument being that while some code may be reran, for small projects this likely is not going to be a bottleneck. 
+
+## Overcoming Technical Debt
+
+The argument against the workflow manager as above is the whole reason `drake` was developed: For non-trivial or long running processes, simply rerunning all the steps is unfeasible.
+
+> With the `code_to_function()` we can now move these encumbered workflows into the managed plan ecosystem of `drake`.
+
+`code_to_function()` accepts the path to either an R or Rmd script and will generate a function that contains the contents of the script you provided as a function.
+
+
+```{r}
+
+load_data <- code_to_function("01_data.R")
+
+```
+
+By turning these scripts into functions, they can be added into a `drake` plan for workflow management.
+
+
+## Dependencies
+
+An important aspect of drake plans is the capability to track the dependencies of each function and only rerun the parts of the plan that have updated, either in code or input. To take advantage of this feature, we will need to first generate a plan from loading the data to generating a histogram. In a normal non-`drake` workflow, the format might some something along the lines of the following.
+
+```{r eval=FALSE}
+
+source("01_data.R")
+source("02_munge.R")
+source("03_histogram.R")
+
+```
+
+Lets convert all of these into functions via `code_to_function()` and put them into a plan.
+
+```{r}
+
+load_data <- code_to_function("01_data.R")
+do_munge  <- code_to_function("02_munge.R")
+generate_histogram <-code_to_function("03_histogram.R")
+
+simple_plan <- drake_plan(
+  data     = load_data(),
+  munged_data = do_munge(),
+  hist     = generate_histogram()
+)
+
+```
+
+This looks good and dandy, but right now `drake` has no idea which part to run first, and what scripts depend on the outputs of the next!
+
+```{r}
+
+simple_config <- drake_config(simple_plan)
+vis_drake_graph(simple_config)
+
+```
+
+## Building the connections
+
+If we knew what scripts need to be run before others, we can use `targets` to indicate to `drake` in which order scripts need to be evaluated. 
+
+In the case described above, the final script, "03_histogram.R", generates a plot of a histogram using the data generated from "02_munge.R", which in turn relies on "01_data.R" to load the data. "01_data.R" is the first script that needs to be evaluated and does not rely on any other inputs.
```

Comment:
Maybe add an explicit mention that function arguments express this dependency relationship.